src/ssl.c

```diff
@@ -22070,6 +22073,24 @@ WOLFSSL_EVP_PKEY* wolfSSL_X509_get_pubkey(WOLFSSL_X509* x509)
             }
             #endif /* HAVE_ECC */
 
+            /* decode ED25519 key */
+            #ifdef HAVE_ED25519
+            if (key->type == EVP_PKEY_ED25519) {
+                key->ed25519 = wolfSSL_ED25519_new();
+                if (key->ed25519 == NULL) {
+                    wolfSSL_EVP_PKEY_free(key);
+                    return NULL;
+                }
+
+                if (wc_ed25519_import_public((const unsigned char*)key->pkey.ptr,
+                       key->pkey_sz, (ed25519_key*)key->ed25519->internal) != 0) {
+                    WOLFSSL_MSG("wc_ed25519_import_public failed");
+                    wolfSSL_EVP_PKEY_free(key);
+                    return NULL;
+                }
+            }
```

Comment:
You need to set `inSet` and `exSet` appropriately after this section. Please implement a function like `wolfSSL_ED25519_LoadDer` (similar to `wolfSSL_RSA_LoadDer`) to handle all this ie: loading the buffer into the internal structure and then setting the external members and setting the `inSet` and `exSet` fields.